src/Lean/Meta/Order.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Wojciech Różowski
+-/
+
+prelude
+import Lean.Meta.InferType
+import Lean.Meta.PProdN
+import Lean.Meta.AppBuilder
+import Init.Internal.Order.Basic
+
+namespace Lean.Meta
+open Order
+
+def mkInstPiOfInstForall (x : Expr) (inst : Expr) : MetaM Expr := do
```

Comment:
Our future selves will probably be grateful for

```suggestion
/--
Given `inst : CCPO (t[x])` for some FVar `x`, constructs an instance
of the type type; `CCPO (∀ x, t[x])`.
Can handle `CompleteLattice` as well.
-/
def mkInstPiOfInstForall (x : Expr) (inst : Expr) : MetaM Expr := do
  assert! x.isFVar
```

And similar docstrings below.